core/templates/pages/exploration-editor-page/statistics-tab/issues/playthrough-issues.component.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for visualizing issues.
+ */
+
+require(
+  'pages/exploration-editor-page/statistics-tab/issues/' +
+  'early-quit-issue.component.ts');
+require(
+  'pages/exploration-editor-page/statistics-tab/issues/' +
+  'multiple-incorrect-issue.component.ts');
+require(
+  'pages/exploration-editor-page/statistics-tab/issues/' +
+  'cyclic-transitions-issue.component.ts');
+
+require('domain/utilities/url-interpolation.service.ts');
+require('services/playthrough-issues.service.ts');
+
+angular.module('oppia').component('playthroughIssues', {
+  template: require('./playthrough-issues.component.html'),
+  controller: [
+    '$scope', 'PlaythroughIssuesService',
+    function($scope, PlaythroughIssuesService) {
+      var ctrl = this;
+      $scope.getIssueIndex = function(issue) {
+        return $scope.issues.indexOf(issue);
+      };
+
+      $scope.currentIssueIndex = -1;
+
+      $scope.isIssueOnInitDisplay = function(issue) {
+        return $scope.getIssueIndex(issue) === 0;
+      };
+
+      $scope.createIssueNavId = function(issue) {
+        return $scope.getIssueIndex(issue) + 1;
+      };
+
+      $scope.isIssueDisplayed = function() {
+        return $scope.currentIssueIndex > -1;
+      };
+
+      $scope.makeVisible = function(nextIssueIndex) {
+        if ($scope.isIssueDisplayed()) {
+          document.getElementById(
```

Comment:
I'm a bit worried about this -- it seems to me like this should not be looking at the DOM in the first place. Is it possible to rewrite this so that it doesn't do that? I worry a bit that it would make the tests flaky, too.

/cc @brianrodri

Reply:
It should be possible if we add `$scope.isIssueIndexVisible`. We can update `makeVisible` to manipulate that array insead, and have each issue element use the corresponding index with `ng-show`.

Reply:
Sg, thanks @brianrodri for the pointer. @marianazangrossi, just a note, it should be $scope.issueIndexIsVisible if it's a boolean variable.

Reply:
I just have few questions:

- How do I suppose to handle this behavior in an array with only one variable (`$scope.isIssueIndexVisible`)?
- If I'm going to handle each issue element, is that ok to set a property in each item?

I think I didn't understand it completely.

Reply:
Hi @marianazangrossi, I'm sorry, I don't know.  I'll need to defer to @brianrodri to answer your questions!

Reply:
@brianrodri PTAL!

Reply:
Hi @seanlip, me and @brianrodri have discussed about this problem and Brian suggested to keep the DOM manipulation for now, since playthrough-issue is not visible in the application (we can't test it to check if it's working after the changes). Brian said he can clean it up when developing the redesign of playthrough-issue. For now, I've just filed an issue #10108  and referred him.

Reply:
OK sg, thanks for following up! LGTM.